### ISSUE TITLE
Taffy rounding bug can cause UI nodes to expand each update

### ISSUE THREAD

**Issue by ickshonpe** [bevyengine/bevy#8911]
## Bevy version

main  910f984709fb58cddbb73

## What you did

```rust
use bevy::prelude::*;
use bevy::window::WindowResolution;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins.set(WindowPlugin {
            primary_window: Some(Window {
                resolution: WindowResolution::new(800., 600.).with_scale_factor_override(1.5),
                ..Default::default()
            }),
            ..Default::default()
        }))
        .add_systems(Startup, setup)
        .run();
}

fn setup(mut commands: Commands) {
    commands.spawn(Camera2dBundle::default());
    commands
        .spawn(NodeBundle {
            style: Style {
                width: Val::Percent(100.),
                left: Val::Px(1.),
                ..Default::default()
            },
            ..Default::default()
        })
        .with_children(|builder| {
            builder
                .spawn(NodeBundle {
                    style: Style {
                        width: Val::Px(100.),
                        justify_content: JustifyContent::End,
                        ..Default::default()
                    },
                    background_color: Color::BLACK.into(),
                    ..Default::default()
                })
                .with_children(|builder| {
                    builder.spawn(NodeBundle {
                        style: Style {
                            min_width: Val::Px(200.),
                            ..default()
                        },
                        background_color: Color::RED.with_a(0.5).into(),
                        ..Default::default()
                    });
                });
        });
}
```

## What went wrong

https://github.com/bevyengine/bevy/assets/27962798/dd44f597-81a8-4318-99a7-6cb2c195201c

The inner leaf node expands by a physical pixel each time the UI updates.

The problem is with how Taffy rounds the layout coordinates. The rust `round` function rounds away from zero, so x bounds from  -100.5 to 99.5 (width 200) will be rounded to -101 to 100 (width 201), gaining a pixel.

The accumulated error resets after a full update (triggered when the window is resized or the scale factor changes).

**Comment by ickshonpe** [comment]
Replicated in Taffy: https://github.com/DioxusLabs/taffy/issues/501

**Comment by ickshonpe** [comment]
Fixed the example, previously it would only show the bug with window scale factor at specific values like `1.5`. Now it overrides the window scale factor and should work for all setups.

**Comment by nicoburns** [comment]
The Taffy fix for this has now landed in `v0.3.13`
